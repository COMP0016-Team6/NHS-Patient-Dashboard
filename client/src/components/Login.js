import React from "react";
import { useDispatch } from "react-redux";
import { loggedIn, loggedOut } from "../state/action";
import { useInput } from "../useInput";
import { loginUser } from "../api/fetches";

import { toast } from "react-toastify";
import { Grid, Avatar } from '@material-ui/core';
import { useStylesLogin } from "../styles/styles";

const Login = () => {
  const classes = useStylesLogin();
  const dispatch = useDispatch();
  const [email, emailField] = useInput({ placeholder: "email *", properties: {name: "email"} });
  const [password, passwordField] = useInput({ type: "password", placeholder: "password *" });

  const onSubmitForm = async e => {
    e.preventDefault();
    try {
      const parseRes = await loginUser({ email, password });
      const user = parseRes.user;
      if (parseRes.jwtToken) {
        localStorage.setItem("token", parseRes.jwtToken);
        dispatch(loggedIn(user.user_role, user.user_id, user.user_name, user.user_email));
        toast.success("Logged in Successfully");
      } else {
        dispatch(loggedOut());
        toast.error(parseRes);
      }
    } catch (err) {
      console.error(err.message);
    }
  };

  return (
    <Grid container component="main" direction="row" justify="center">
      <Grid item sm={4} md={1} />
      <Grid item xs={12} sm={8} md={5} style={{ margin: 30 }}>
        <div className={classes.paper}>
          <Avatar className={classes.avatar} />
          <h1 className="text-center">Login</h1>
          <form className={classes.form} onSubmit={onSubmitForm}>
            {emailField}
            {passwordField}
            <button type="submit" className="btn btn-success btn-block mt-5">Submit</button>
          </form>
        </div>
      </Grid>
      <Grid item xs={false} sm={4} md={4} className={classes.image} style={{ marginLeft: 30 }} />
    </Grid>
  );
};

export default Login;
