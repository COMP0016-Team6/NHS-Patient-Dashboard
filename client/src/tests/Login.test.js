import React from "react";
import Login from "../components/Login";
import renderer from "react-test-renderer";
import { wrapProvider } from "./util";
import { BrowserRouter } from 'react-router-dom';

it ("login snapshot matches", () => {
  const tree = renderer.create(wrapProvider(<BrowserRouter><Login /></BrowserRouter>));
  expect(tree).toMatchSnapshot();
});
