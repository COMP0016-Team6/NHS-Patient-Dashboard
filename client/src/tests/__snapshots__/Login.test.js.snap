// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login snapshot matches 1`] = `
<main
  className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-1"
  />
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-5"
    style={
      Object {
        "margin": 30,
      }
    }
  >
    <div
      className="makeStyles-paper-2"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiAvatar-fallback"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
      <h1
        className="text-center"
      >
        Login
      </h1>
      <form
        className="makeStyles-form-4"
        onSubmit={[Function]}
      >
        <div
          className="MuiFormControl-root MuiTextField-root form-control my-3"
          maxLength={null}
          style={
            Object {
              "color": "#61dafb",
            }
          }
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
          >
            email *
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              name="email"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root form-control my-3"
          maxLength={null}
          style={
            Object {
              "color": "#61dafb",
            }
          }
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
          >
            password *
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="password"
              value=""
            />
          </div>
        </div>
        <button
          className="btn btn-success btn-block mt-5"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
  <div
    className="MuiGrid-root makeStyles-image-3 MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-4"
    style={
      Object {
        "marginLeft": 30,
      }
    }
  />
</main>
`;
